### PR TITLE
add required storage class for Azure example

### PR DESCRIPTION
**Pull request by tybritten**
Without a StorageClass provisioner for Azure, this sets the storage class to the default for etcd in AKS

### CODE REVIEW

**Comment by chainlink** [comment]
Nice find! There'a few things here I think
- It looks like we create an invalid storage class manifest by default on azure. We should update the default to just not render the storage class if a custom one hasn't been defined. If storage class is omitted from the statefulset/deployment, it will just assume default.
- It's actually been recently discovered we need ~1K IOPS for etcd for pach performance. So we should maybe take this opportunity to define a storage class for azure. It looks pretty straight forward from https://dev.to/azure/how-to-use-custom-storage-classes-in-kubernetes-3jo4

Ref:
- https://azure.microsoft.com/en-ca/pricing/details/managed-disks/
- https://docs.microsoft.com/en-us/azure/aks/azure-disks-dynamic-pv

**Comment by tybritten** [comment]
Good Call. I'll make those changes

**Comment by tybritten** [comment]
Good Call. I'll make those changes

**Comment by chainlink** [comment]
Nice! looking good. Couple last things:
- I think IOPS is tied to disk size in azure, so we might want to set a min there
- Do you mind changing https://github.com/pachyderm/helmchart/blob/6f662cbe312fa615391f448f11b63309551e0012/pachyderm/templates/etcd/storageclass.yaml#L5
to 
```
{{- if or (eq .Values.pachd.storage.backend "AMAZON") (eq .Values.pachd.storage.backend "GOOGLE") (eq .Values.pachd.storage.backend "MICROSOFT")}}
```
So we don't get other instances of it accidentally being rendered?

**Comment by tybritten** [comment]
@chainlink ok, I tried a few different ways, but I think nested ifs are the most readable to users

**Comment by chainlink** [comment]
Looking good, thoughts @robert-uhl ?

File: test/microsoft_test.go [hunk 1]
```diff
@@ -20,6 +21,7 @@ func TestMicrosoft(t *testing.T) {
 			&helm.Options{
 				SetStrValues: map[string]string{
 					"pachd.storage.backend":             "MICROSOFT",
+					"etcd.storage.backend":              "MICROSOFT",
```
Comment (robert-uhl) [inline test/microsoft_test.go:24]: Is `etcd.storage.backend` referred to anywhere? I see that the `etcd` manifests refer to `pachd.storage.backend`.

@chainlink, what do you think: should storage be configurable separately for `pachd` and `etcd` (more flexible) or should it be global (simpler)?

Comment (tybritten) [reply test/microsoft_test.go:24]: great point. IMHO it should probably be global

Comment (chainlink) [reply test/microsoft_test.go:24]: Agree with global, we can break them out when there is a request for it.

It does raise an interesting question though. Right now we're conflating object store with cloud provider (which addresses almost all cases, I've never heard of a AWS cluster hitting a GCP Bucket for example) I'm not sure we want to continue to do that, or if we do, we should make it more explicit (by moving it out of `pachd` perhaps?)

**Review by chainlink — APPROVED** [review]

**Pull request merged**